src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -1138,23 +1138,43 @@ bool SlaterDetBuilder::readDetListH5(xmlNodePtr cur,
               << " Optimized coefficients were substituted to the original set of coefficients." << std::endl;
   }
 
-  std::vector<Matrix<int64_t>> temps;
+  std::vector<Matrix<uint64_t>> temps;
   for (int grp = 0; grp < nGroups; grp++)
   {
-    Matrix<int64_t> tmp(ndets, N_int);
+    Matrix<uint64_t> tmp(ndets, N_int);
     temps.push_back(tmp);
     std::string ci_str;
 
-    if (!hin.readEntry(temps[grp], "CI_" + std::to_string(grp)))
+    bool is_correct_type = false;
+    std::string ds_tag   = "CI_" + std::to_string(grp);
+    is_correct_type      = hin.is_dataset_of_type<uint64_t>(ds_tag);
+    if (!is_correct_type)
     {
       //for backwards compatibility
       if (grp == 0)
-        hin.read(temps[grp], "CI_Alpha");
+      {
+        ds_tag          = "CI_Alpha";
+        is_correct_type = hin.is_dataset_of_type<uint64_t>(ds_tag);
+      }
       else if (grp == 1)
-        hin.read(temps[grp], "CI_Beta");
-      else
-        APP_ABORT("Unknown HDF5 CI format");
+      {
+        ds_tag          = "CI_Beta";
+        is_correct_type = hin.is_dataset_of_type<uint64_t>(ds_tag);
+      }
     }
+
+    if (!is_correct_type)
```

Comment:
Why repeat the same check on line 1151?

Reply:
The first check just checks that the numbered "CI_X" tag wasn't found (e.g. "CI_0"). For historical reasons we have to check if "CI_Alpha/Beta" exists as well. So if `!correct_type` happens the first time, we still have to check again.